app/code/community/Algolia/Algoliasearch/Helper/Entity/Producthelper.php

```diff
@@ -1041,6 +1043,44 @@ public function getObject(Mage_Catalog_Model_Product $product)
         return $customData;
     }
 
+    /**
+     * Returns all parent product IDs, e.g. when simple product is part of configurable or bundle
+     *
+     * @param array $productIds
+     * @return array
+     */
+    public function getParentProductIds(array $productIds)
+    {
+        $parentIds = array();
+        foreach ($this->getCompositeTypes() as $typeInstance) {
+            $parentIds = array_merge($parentIds, $typeInstance->getParentIdsByChild($productIds));
+        }
+        return $parentIds;
+    }
+
+    /**
+     * Returns composite product type instances
+     *
+     * @return Mage_Catalog_Model_Product_Type[]
+     *
+     * @see Mage_Catalog_Model_Resource_Product_Flat_Indexer::getProductTypeInstances()
+     */
+    private function getCompositeTypes()
+    {
+        if ($this->compositeTypes === null) {
+            /** @var Mage_Catalog_Model_Product $productEmulator */
+            $productEmulator = Mage::getModel('catalog/product');
+            /** @var Mage_Catalog_Model_Product_Type $productType */
```

Comment:
Can you leave a blank line between the `$productEmulator = ...` and `/** @var ...`? :)

Reply:
Will add spacing there!